src/QMCHamiltonians/StressPBC.h

```diff
@@ -0,0 +1,111 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Raymond Clay III, j.k.rofling@gmail.com, Lawrence Livermore National Laboratory
+//                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+//                    Yubo Paul Yang, yyang173@illinois.edu, University of Illinois Urbana-Champaign
+//
+// File created by: Raymond Clay III, j.k.rofling@gmail.com, Lawrence Livermore National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_STRESSPBC_HAMILTONIAN_H
+#define QMCPLUSPLUS_STRESSPBC_HAMILTONIAN_H
+#include "QMCHamiltonians/ForceBase.h"
+#include "LongRange/LRCoulombSingleton.h"
+#include "OhmmsPETE/SymTensor.h"
+
+namespace qmcplusplus
+{
+struct StressPBC : public OperatorBase, public ForceBase
+{
+  typedef LRCoulombSingleton::LRHandlerType LRHandlerType;
+
+  SymTensor<RealType, OHMMS_DIM> stress_ee_const;
+  SymTensor<RealType, OHMMS_DIM> stress_eI_const;
+
+  // need wave function for kinetic stress
+  TrialWaveFunction& Psi;
+
+  ///source particle set
+  ParticleSet& PtclTarg;
+  ParticleSet& PtclA;
+  ///long-range Handler
+  LRHandlerType* AA;
+  ///locator of the distance table
+  const int ei_table_index;
+  /// e-e table ID
+  const int ee_table_index;
+  const int ii_table_index;
+  ///number of species of A particle set
+  int NumSpeciesA;
+  ///number of species of B particle set
+  int NumSpeciesB;
+  ///number of particles of A (classical, e.g. ions)
+  int NptclA;
+  ///number of particles of B (quantum, e.g. electrons)
+  int NptclB;
+
+  ///number of particles per species of A
+  std::vector<int> NofSpeciesA;
+  ///number of particles per species of B
+  std::vector<int> NofSpeciesB;
+  ///Zat[iat] charge for the iat-th particle of A
+  std::vector<RealType> Zat;
+  ///Qat[iat] charge for the iat-th particle of B
+  std::vector<RealType> Qat;
+  ///Zspec[spec] charge for the spec-th species of A
+  std::vector<RealType> Zspec;
+  ///Qspec[spec] charge for the spec-th species of B
+  std::vector<RealType> Qspec;
+  //Constructor
+  bool firstTimeStress;
+  StressPBC(ParticleSet& ions, ParticleSet& elns, TrialWaveFunction& Psi);
+
+  Return_t evaluate(ParticleSet& P);
```

Comment:
Please use override keyword to mark all the functions overriding virtual functions. Once you introduce one, clang warns you all the rest. So check with clang is helpful.

Reply:
override keywords added